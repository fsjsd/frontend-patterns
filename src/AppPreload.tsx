import React, { PropsWithChildren, useLayoutEffect } from 'react';
import { PageHeader } from './ux/PageHeader';
import { ThemeProvider } from 'styled-components';
import theme from './ux/theme';
import { SectionMain, SiteContainer, FixedApp, HeaderBrand } from './AppStyles';
import { NavigationDrawer, NavigationScrollArea } from './ux/NavigationContainerStyles';
import { TombStoneContent } from './ux/designsystem/TombStones';

// see: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/59765
const ThemeProviderFixed = ThemeProvider as unknown as React.FC<PropsWithChildren<{ theme: typeof theme }>>;


function AppPreload() {

  useLayoutEffect(() => {
    const documentContent = document.querySelector("#root")?.innerHTML;
    // Dump HTML ...
    console.log(documentContent);

    let css = "";
    // Dump Stylesheets ...
    Array.from(document.styleSheets).forEach((sheet) => {
      const cssRules = Array.from(sheet.cssRules).map(rule => rule.cssText).join("\n");
      css += cssRules;
      // console.log("CSS:");
      // console.log(cssRules)
    });
    console.log(css);
  }, []);

  const handleMenuClick = () => {
    // no op
  }

  return (
    <ThemeProviderFixed theme={theme}>
      <FixedApp>
        <SiteContainer>
          <NavigationDrawer visible={false}>
            <HeaderBrand />
            <NavigationScrollArea role="navigation" />
          </NavigationDrawer>
          <SectionMain>
            <PageHeader showLinks={false} onMenuClick={handleMenuClick} />
            <div style={{ padding: "15px" }}>
              <TombStoneContent />
            </div>
          </SectionMain>
        </SiteContainer>
      </FixedApp>
    </ThemeProviderFixed>
  );
}

export default AppPreload;
