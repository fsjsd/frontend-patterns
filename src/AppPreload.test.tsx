import React from 'react';
import { render } from '@testing-library/react';
import AppPreload from './AppPreload';
import { toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

/**
 * Renders content for static html. 
 * 
 * Copy the output directly from the test snaphost into public/index.html,
 * note the CSS will require style tags and the HTML will require closing.
 * CSS keyframes will need to be manually copied.
 * 
 * IMPORTANT: Requires jest-styled-components to render CSS in usable
 * form.
 */

describe('AppPreload tests', () => {
  test('renders static preload html', () => {
    const { container } = render(<AppPreload />);
    expect(container.firstChild).toMatchSnapshot();
  });
})
