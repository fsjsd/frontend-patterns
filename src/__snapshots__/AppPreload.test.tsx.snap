// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPreload tests renders static preload html 1`] = `
.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 3.5rem;
  height: 3.5rem;
  padding: 0rem 1rem;
  background-color: #0076D1;
  color: white;
}

.c7 {
  font-weight: 400;
  font-size: 1.25rem;
  line-height: inherit;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  color: inherit;
  margin: 0;
  padding: 0;
}

.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c1 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-family: Roboto,Arial,Helvetica;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c5 {
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3.5rem;
  min-height: 3.5rem;
  padding: 0rem 1rem;
  background-color: #007fe0;
  color: #fff;
  font-size: 1px;
  visibility: hidden;
  pointer-events: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  pointer-events: none;
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: 100;
  -webkit-transition: -webkit-transform 0.2s ease-in-out;
  -webkit-transition: transform 0.2s ease-in-out;
  transition: transform 0.2s ease-in-out;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c4 {
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow-y: auto;
  overflow-x: hidden;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
  -webkit-overflow-scrolling: touch;
  border-left: solid 1px rgb(218,231,241);
  background-color: rgb(246,250,253);
  pointer-events: all;
  padding-bottom: env(safe-area-inset-bottom);
}

.c8 {
  margin-bottom: 1rem;
  width: 100%;
  height: 1rem;
  display: block;
  border-radius: 2px;
  background: #eee;
  background: linear-gradient(110deg,#ececec 8%,#e8e8e8 18%,#ececec 33%);
  background-size: 200% 100%;
  -webkit-animation: 1.25s dBsxNk linear infinite;
  animation: 1.25s dBsxNk linear infinite;
}

@media (min-width:768px) {
  .c3 {
    visibility: visible;
    pointer-events: all;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    margin-top: 0px;
    position: inherit;
    z-index: inherit;
    width: 240px;
  }
}

@media (min-width:768px) {
  .c4 {
    width: 240px;
  }
}

<div
  class="c0"
  id="root"
>
  <div
    class="c1"
  >
    <section
      class="c2"
    >
      <header
        class="c3"
      />
      <div
        class="c4"
        role="navigation"
      />
    </section>
    <section
      class="c5"
    >
      <header
        class="c6"
        role="banner"
      >
        <h1
          class="c7"
        >
          <b>
            Front-End
          </b>
           
          Design Patterns
        </h1>
      </header>
      <div
        style="padding: 15px;"
      >
        <div
          class="c8"
          style="height: 2rem; width: 45%;"
        />
        <div
          class="c8"
        />
        <div
          class="c8"
        />
        <div
          class="c8"
        />
      </div>
    </section>
  </div>
</div>
`;
